chain/client/src/client.rs

```diff
@@ -237,6 +237,15 @@ impl Client {
             return Ok(None);
         }
 
+        // Process stored partial encoded chunks
```

Comment:
A better place to process them would be when the previous block is accepted, not when the new one is generated. Doing it here doesn't cover multiple cases:

1. The block producer tracks some shards, unknown to the participant sending the `PartialEncodedChunkMessage`. For such a block producer just accepting the `PartialEncodedChunkmesasge` won't be enough, they need to request more parts (but it's too late here, because they are already producing the block, so the chunk will just be skipped).
2. A height is skipped. `height_created` is always equal to the prev block height + 1, but is not always equal to the height at which the chunk will be included. So if you process stored chunks on accepting the previous block, you are guaranteed to process all of them even if some height will be skipped in the future. But here you'd skip all the chunks with `height_created = 7` when creating a block for height 8, even if the block at height 7 was never created.